rollouts/api/v1alpha1/remotesync_types.go

```diff
@@ -23,21 +23,22 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// RemoteRootSyncSpec defines the desired state of RemoteRootSync
-type RemoteRootSyncSpec struct {
+// RemoteSyncSpec defines the desired state of RemoteSync
+type RemoteSyncSpec struct {
```

Comment:
After going through all the changed code, I realized separating the two would have also simplified the review :) because naming change has hijacked the real changes.

Reply:
I split out the name change into its own PR: https://github.com/GoogleContainerTools/kpt/pull/3857

I will update this PR to rebase on top of https://github.com/GoogleContainerTools/kpt/pull/3857 and address your other comments.